src/mongocxx/database.cpp

```diff
@@ -157,5 +164,12 @@ collection database::collection(stdx::string_view name) const {
     return mongocxx::collection(*this, name);
 }
 
+const std::unique_ptr<database::impl>& database::get_impl() const {
+    if(!_impl) {
+        throw operation_exception{make_error_code(error_code::k_database_uninitialized)};
```

Comment:
I meant that get_impl() is called by a variety of database methods, and I'm not sure all of these methods would want to throw an operation_exception. Probably for databases, this is okay, but if we do the same thing for client and collection, I don't know whether we would want all methods to throw operation_exception, or have some of them throw query_exception, etc.